manifests/init.pp

```diff
@@ -151,31 +151,44 @@
     require => Package[$packagename],
   }
 
-  # Only debian and ubuntu have this style of defaults for startup.
-  case $::osfamily {
-    'Debian': {
-      file{ '/etc/default/isc-dhcp-server':
-        ensure  => present,
-        owner   => 'root',
-        group   => 'root',
-        mode    => '0644',
-        before  => Package[$packagename],
-        notify  => $service_notify_real,
-        content => template('dhcp/debian/default_isc-dhcp-server'),
-      }
+  if $::service_provider == 'systemd' {
+    file { '/etc/systemd/system/dhcpd.service':
+      ensure  => present,
+      owner   => 'root',
+      group   => 'root',
+      mode    => '0644',
+      before  => Package[$packagename],
+      notify  => Service[$servicename],
+      content => template('dhcp/dhcpd.service'),
```

Comment:
you can use epp if you like that more

Reply:
We need a notify here for systemctl reload. The resource should come from https://github.com/camptocamp/puppet-systemd#overview. Could you please replace the file resource with ::systemd::unit_file ?